internal/test/framework/framework.go

```diff
@@ -159,6 +174,81 @@ func (f *TestFramework) getKubeClient() error {
 	return nil
 }
 
+// GetNode returns a pointer to the node object associated with the external IP provided
+func (f *TestFramework) GetNode(externalIP string) (*v1.Node, error) {
+	var matchedNode *v1.Node
+
+	nodes, err := f.K8sclientset.CoreV1().Nodes().List(metav1.ListOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("could not get list of nodes")
+	}
+	if len(nodes.Items) == 0 {
+		return nil, fmt.Errorf("no nodes found")
+	}
+
+	// Find the node that has the given IP
+	for _, node := range nodes.Items {
+		for _, address := range node.Status.Addresses {
+			if address.Type == "ExternalIP" && address.Address == externalIP {
+				matchedNode = &node
+				break
+			}
+		}
+		if matchedNode != nil {
+			break
+		}
+	}
+	if matchedNode == nil {
+		return nil, fmt.Errorf("could not find node with IP: %s", externalIP)
+	}
+	return matchedNode, nil
+}
+
+// k8sVersionToOpenShiftVersion converts a Kubernetes minor version to an OpenShift version in format
+// "major.minor". This function works under the assumption that for OpenShift 4, every OpenShift minor version increase
+// corresponds with a kubernetes minor version increase
+func k8sVersionToOpenShiftVersion(k8sMinorVersion int) (string, error) {
```

Comment:
nit: Are we sure this is the only way to get OpenShift cluster version?

Reply:
I could not find another way and was told that this is the version for all PR builds.

Reply:
Add a TODO, if this needs more investigation, create a Jira story and I'll LGTM.

Reply:
Add a TODO for what? What would you like to see done?

Reply:
Had an offline chat with @sebsoto and he thinks this doesn't need more investigation.